services/dyndata.ts

```diff
@@ -6,6 +10,42 @@ const api = $fetch.create({
   baseURL: BASE_URL,
 })
 
+const apiInstanceByPrefix = (prefix: Prefix) => {
+  const endpoint: string = getChainEndpointByPrefix(prefix) || ''
+  return ApiFactory.useApiInstance(endpoint)
+}
+
+const lastIdAssethub = async (prefix: Prefix, collectionId: number) => {
+  const api = await apiInstanceByPrefix(prefix)
+  const ids = await api.query.nfts.item.keys(collectionId)
+  const ints = ids.map(id => Number.parseInt(id.toHuman()?.[1]?.replaceAll(',', '') || '0'))
+  const biggestId = Math.max(...ints)
+  return biggestId
+}
+
+// store latest IDs for each collection
+const latestIdsMap = new Map<string, number>()
+
+export const generateIdAssethub = async (collectionId: number, prefix?: Prefix) => {
```

Comment:
kinda verbose but something along those lines 

```suggestion
export const generateBackwardsCompatibilityId = async (collectionId: number, prefix?: Prefix) => {
```